packages/aws-cdk-lib/aws-cloudfront/lib/realtime-log-config.ts

```diff
@@ -0,0 +1,106 @@
+import { Construct } from 'constructs';
+import { CfnRealtimeLogConfig } from './cloudfront.generated';
+import * as iam from '../../aws-iam';
+import * as kinesis from '../../aws-kinesis';
+import { IResource, Resource } from '../../core';
+
+/**
+ * Represents the endpoints available for targetting within a realtime log config resource
+ */
+export abstract class Endpoint {
+  public static fromKinesisStream(stream: kinesis.IStream, role: iam.IRole): Endpoint {
```

Comment:
* The role could be made optional and created by the construct
* The IAM permissions should be automatically configured by the construct (https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/real-time-logs.html#understand-real-time-log-config-iam-role)

Reply:
I did it this way as to auto-generate we would have to alter it to also include KMS key (for encrypted streams)? i.e.
```
  public static fromKinesisStream(stream: kinesis.IStream, role?: iam.IRole, kmsKey?: kms.Key): Endpoint {
```

Reply:
I'll take a look thanks!